sdn_tests/testplans/gNMI-GET-Modes-testplan.md

```diff
@@ -0,0 +1,868 @@
+```
+
+
+   </td>
+
+### Test #1: GET Config at the subtree level
+Condition to test:
+```
+path: { origin: "openconfig" elem: { name:"interfaces" } elem: { name:"interface" key: { key: "name" value: "EthernetX" } } } type: DataType_CONFIG
+```
+
+Validation: 
+Verify that only the config subtrees and their leaf paths are returned:
+
+Example:
+```
+request:
+'prefix: { origin: "openconfig", target: "<Target>" }, path: { elem: { name:"interfaces" } elem: { name:"interface" key: { key: "name" value: "EthernetX" } } } type: 1'
+
+response:
+  update: <
+    path: <
+      elem: <
+        name: "interfaces"
+      >
+      elem: <
+        name: "interface"
+        key: <
+          key: "name"
+          value: "EthernetX"
+        >
+      >
+    >
+    val: <
+      json_ietf_val: "{\"openconfig-interfaces:interface\":[{\"config\":{\"enabled\":true,\"mtu\":<MTU>,\"name\":\"EthernetX\",\"type\":\"iana-if-type:ethernetCsmacd\"},\"name\":\"EthernetX\",\"subinterfaces\":{\"subinterface\":[{\"config\":{\"index\":0},\"index\":0,\"openconfig-if-ip:ipv4\":{\"addresses\":{\"address\":[{\"config\":{\"ip\":\"<IP_ADD>\",\"prefix-length\":XX},\"ip\":\"<IP_ADD>\"}]},\"config\":{\"enabled\":false}},\"openconfig-if-ip:ipv6\":{\"config\":{\"enabled\":false}}}]}}]}"
+    >
+  >
+>
+```
+
+### Test #2: GET Config at the root level
+Condition to test:
+```
+path: { origin: "openconfig" elem: { name:"/" } } type: DataType_CONFIG
+```
+
+Validation: 
+Verify that only the config subtrees and their leaf paths are returned. 
+
+### Test #3: GET Config at the leaf level
+Condition to test:
+```
+path: { origin: "openconfig" elem: { name:"interfaces" } elem: { name:"interface" key: { key: "name" value: "EthernetX" } }  elem: { name:"config" } elem: { name:"mtu" } } type: DataType_CONFIG
+```
+
+Validation: 
+Verify that /interfaces/interface/[name=EthernetX]/config/mtu value (scalar value) is returned.
+
+<td style="background-color: #fafafa">
+
+Example:
+```
+request:
+'prefix: { origin: "openconfig", target: "<target>" }, path: { elem: { name:"interfaces" } elem: { name:"interface" key: { key: "name" value: "EthernetX" } } elem: { name:"config" }  elem : {name:"mtu"} } type: 1'
+
+response:
+  update: <
+    path: <
+      elem: <
+        name: "interfaces"
+      >
+      elem: <
+        name: "interface"
+        key: <
+          key: "name"
+          value: "EthernetX"
+        >
+      >
+      elem: <
+        name: "config"
+      >
+      elem: <
+        name: "mtu"
+      >
+    >
+    val: <
+      json_ietf_val: "{\"openconfig-interfaces:mtu\":<MTU>}"
+    >
+  >
+>
+```
+
+
+   </td>
+
+### Test #4: Negative test case - GET Config type with state/operational subtree
+Condition to test: Config type with state/operational subtree
+
+```
+path: { origin: "openconfig" elem: { name:"interfaces" } elem: { name:"interface" key: { key: "name" value: "EthernetX" } elem: { name:"state"} } } type: DataType_CONFIG
+```
+
+Validation: should return empty subtree
```

Comment:
Should not we follow the gnmi spec in these negative cases and return NOT_FOUND error: https://github.com/sonic-net/sonic-mgmt/pull/12475/files#diff-379e9779ab5abad2e67342176931d95671bf40165bd29b9d70e0af3781e85925R325